### ISSUE TITLE
Querying a new parquet table fails with "A group type can not be empty. Parquet does not support empty group without leaves. Empty group: hive_schema"

### ISSUE THREAD

**Issue by lhofhansl** [prestosql/presto#201]
This started to happen fairly recently... Presto-302. (When I get time I will double check with 301 and 300)

Transcript:
```
presto> create table hive.test.data (pk1 integer, v1 integer) WITH (format = 'PARQUET');
CREATE TABLE

presto> insert into hive.test.data values(cast (rand()*10 as integer), cast (rand()*100 as integer));
INSERT: 1 row

Query 20190210_063547_00034_3ezmi, FINISHED, 1 node
Splits: 35 total, 35 done (100.00%)
0:01 [0 rows, 0B] [0 rows/s, 0B/s]

presto> select count(*) from hive.test.data;

Query 20190210_063553_00035_3ezmi, FAILED, 1 node
Splits: 18 total, 0 done (0.00%)
0:00 [0 rows, 0B] [0 rows/s, 0B/s]

Query 20190210_063553_00035_3ezmi failed: Error opening Hive split hdfs://localhost:9000/hive_catalog/test/data/20190210_063547_00034_3ezmi_c940c42f-900b-4947-a554-19b9840fcb57 (offset=0, length=350): A group type can not be empty. Parquet does not support empty group without leaves. Empty group: hive_schema
```

hive.properties:
```
connector.name=hive-hadoop2
hive.metastore=file
hive.metastore.catalog.dir=hdfs://localhost:9000//hive_catalog
hive.metastore.user=lars
hive.allow-drop-table=true
```

Same happens when I run hive metastore, with:
```
connector.name=hive-hadoop2
hive.metastore.uri=thrift://localhost:9083
hive.allow-drop-table=true
```

**Comment by lhofhansl** [comment]
I can read the very same Parquet file with Spark, so it must be related to the Presto Parquet reader.

**Comment by lhofhansl** [comment]
Confirmed that this is not a problem Presto-300

**Comment by dain** [comment]
@lhofhansl can you run the CLI with the `--debug` option, and add the stack trace to the description of this issue?

**Comment by dain** [comment]
Actually, looks like it is an issue with Parquet version used in Presto. https://stackoverflow.com/a/48048311/2958751

**Comment by lhofhansl** [comment]
Here's the full exception. (My apologies for not attaching that in the beginning.)

```
Query 20190211_024055_00000_htvpd failed: Error opening Hive split hdfs://localhost:9000/hive_catalog/test/data/20190210_063547_00034_3ezmi_c940c42f-900b-4947-a554-19b9840fcb57 (offset=0, length=350): A group type can not be empty. Parquet does not support empty group without leaves. Empty group: hive_schema
io.prestosql.spi.PrestoException: Error opening Hive split hdfs://localhost:9000/hive_catalog/test/data/20190210_063547_00034_3ezmi_c940c42f-900b-4947-a554-19b9840fcb57 (offset=0, length=350): A group type can not be empty. Parquet does not support empty group without leaves. Empty group: hive_schema
        at io.prestosql.plugin.hive.parquet.ParquetPageSourceFactory.createParquetPageSource(ParquetPageSourceFactory.java:232)
        at io.prestosql.plugin.hive.parquet.ParquetPageSourceFactory.createPageSource(ParquetPageSourceFactory.java:120)
        at io.prestosql.plugin.hive.HivePageSourceProvider.createHivePageSource(HivePageSourceProvider.java:161)
        at io.prestosql.plugin.hive.HivePageSourceProvider.createPageSource(HivePageSourceProvider.java:95)
        at io.prestosql.spi.connector.classloader.ClassLoaderSafeConnectorPageSourceProvider.createPageSource(ClassLoaderSafeConnectorPageSourceProvider.java:44)
        at io.prestosql.split.PageSourceManager.createPageSource(PageSourceManager.java:56)
        at io.prestosql.operator.TableScanOperator.getOutput(TableScanOperator.java:239)
        at io.prestosql.operator.Driver.processInternal(Driver.java:379)
        at io.prestosql.operator.Driver.lambda$processFor$8(Driver.java:283)
        at io.prestosql.operator.Driver.tryWithLock(Driver.java:675)
        at io.prestosql.operator.Driver.processFor(Driver.java:276)
        at io.prestosql.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1075)
        at io.prestosql.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
        at io.prestosql.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:483)
        at io.prestosql.$gen.Presto_302____20190211_024038_1.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: org.apache.parquet.schema.InvalidSchemaException: A group type can not be empty. Parquet does not support empty group without leaves. Empty group: hive_schema
        at org.apache.parquet.schema.GroupType.<init>(GroupType.java:92)
        at org.apache.parquet.schema.GroupType.<init>(GroupType.java:48)
        at org.apache.parquet.schema.MessageType.<init>(MessageType.java:50)
        at io.prestosql.plugin.hive.parquet.ParquetPageSourceFactory.createParquetPageSource(ParquetPageSourceFactory.java:172)
        ... 17 more
```

**Comment by lhofhansl** [comment]
> Actually, looks like it is an issue with Parquet version used in Presto. https://stackoverflow.com/a/48048311/2958751

I think that would not explain why it worked in Presto-300.
I looked (very) superficially through the commits between 300 and 301. These seem suspicious at least:
db316f1777cb3ba7657127e31568b18e4b3e9ee3
f800115d0111822e9e290ae968986b4e73085f25

(I won't have time in the next few days to do a bisect)

**Comment by electrum** [comment]
This problem occurs when a query reads from Parquet but does not select any columns (only positions), so it happens for a query like `SELECT count(*) FROM t`.

It is fixed by #203. I was able to write a simple unit test for it. Thank you for providing clear instructions on reproducing it.

**Comment by lhofhansl** [comment]
So in the end this is cause by 238f4b6bb1ac98130cff7bee88d5e622e5a2808f?
